JASP-Engine/JASP/R/descriptives.R

```diff
@@ -1427,4 +1455,89 @@ Descriptives <- function(jaspResults, dataset, options) {
     return(TRUE)
   else
     return(FALSE)
-}
\ No newline at end of file
+}
+
+.Xbarchart <- function(jaspResults, dataset, options, variables){
+
+  ready <- (length(options$variables) > 0)
+
+  if (ready && is.null(jaspResults[["Xbarchart"]]))
+    .XbarPlotDescriptives(jaspResults, dataset, options, ready)
+
+  return()
+}
+
+.XbarPlotDescriptives <- function(jaspResults, dataset, options, ready){
+
+  XbarPlot <- createJaspPlot(title = "X bar chart", width=500, aspectRatio=1)
+  jaspResults[["Xbarchart"]] <- XbarPlot
+
+  if (!ready)
+    return()
+
+  .XbarFillPlotDescriptives(XbarPlot, dataset, options)
+
+  return()
+}
```

Comment:
you can merge these functions if they're this small:
```suggestion
.Xbarchart <- function(jaspResults, dataset, options, variables){

  ready <- (length(options$variables) > 0)

  if (!ready || !is.null(jaspResults[["Xbarchart"]]))
    return()
    
  XbarPlot <- createJaspPlot(title = "X bar chart", width=500, aspectRatio=1)
  jaspResults[["Xbarchart"]] <- XbarPlot

  .XbarFillPlotDescriptives(XbarPlot, dataset, options)

  return()
}
```
that would also make it clear that you're having twice `if (ready)` in there.